distributed/core.py

```diff
@@ -645,7 +645,7 @@ async def send_recv(comm, reply=True, serializers=None, deserializers=None, **kw
             response = await comm.read(deserializers=deserializers)
         else:
             response = None
-    except EnvironmentError:
+    except (EnvironmentError, CommClosedError):
```

Comment:
Note here that `CommClosedError` was not being handled in this function, but even the TCP case may raise that in both [`read`](https://github.com/dask/distributed/blob/main/distributed/comm/tcp.py#L190) and [`write`](https://github.com/dask/distributed/blob/main/distributed/comm/tcp.py#L233) methods.

Reply:
Though it seems setting `force_close` impacts the `finally` case below. Are we sure we can remove this?

Reply:
When you say "remove", what exactly are you referring to? Here I'm adding a new previously uncatched exception, which may affect UCX. I was actually comparing to conditions that may happen in TCP comms as well, then I saw this was uncatched but is totally valid (i.e., can occur).